swift-generator/src/main/resources/templates/java/common.st

```diff
@@ -333,3 +336,59 @@ public String toString()
         .toString();
 }
 >>
+
+_union_field_constructor(context, field) ::= <<
+@ThriftConstructor
+public <context.javaName>(final <field.javaType> <field.javaName>) {
+    this.value = <field.javaName>;
+    this.id = <field.id>;
+    this.name = "<field.name>";
+}
+>>
+
+_union_constructor(context) ::= <<
+
+private Object value;
+private int id = -1;
+private String name;
+
+public <context.javaName>() {
+}
+
+<if(context.hasUniqueFieldTypes)>
```

Comment:
This `if` (together with `getHasUniqueFieldTypes()`) is the only change since the prev version, right? In this case, lgtm